roles/ssh-hardening/templates/openssh.conf.j2

```diff
@@ -0,0 +1,111 @@
+# {{ansible_managed}}
+
+# This is the ssh client system-wide configuration file.
+# See ssh_config(5) for more information on any settings used. Comments will be added only to clarify why a configuration was chosen.
+#
+# Basic configuration
+# ===================
+
+# Address family should always be limited to the active network configuration.
+AddressFamily {{ 'any' if network_ipv6_enable else 'inet' }}
```

Comment:
Although I prefer `ternary` as well (for consistency with the rest of Trellis), I refrained from implementing Trellis coding conventions, trying leave the original `dev-sec/ansible-ssh-hardening` role as intact as possible for simpler comparison with upstream.

If we choose to apply Trellis conventions, we could consider the following:
1. use `ternary` filter in place of pythonic if/else, as you suggested
2. break list var values out into multiline yaml, e.g., instead of `var: ['val', 'val2']`
3. add space around var names, e.g.,  `{{port}}` becomes `{{ port }}`
4. align the various `# ssh` and `# sshd` comments in `defaults/main.yml`
5. edit code comments for succinctness, grammar, capitalization consistency
6. either parameterize or remove the `IdentityFile` entries in `openssh.conf.j2` template
7. maybe simplify a handful of templating chunks using this pattern:
```diff
-{% if ssh_allow_users -%}
-AllowUsers {{ssh_allow_users}}
-{% endif %}
+AllowUsers {{ ssh_allow_users | default('')}}
```
I believe it's ok -- and has the same effect -- to have empty parameters, e.g., to fall back to default of empty string. I could test it.